File: 1018.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& A) {
        vector<bool>res;
        int currentRear = 0;
        for(int i = 0;i<A.size();i++){
            currentRear = (currentRear * 2 + A[i]) % 10;
            if(currentRear == 0 || currentRear == 5)
                res.push_back(true);
            else
                res.push_back(false);
        }
        return res;
    }
};

File: 1018.java
class Solution {
    public List<Boolean> prefixesDivBy5(int[] A) {
        List<Boolean> res = new ArrayList<>();
        int currentRear = 0;
        for(int i = 0;i < A.length;i++){
            currentRear = (currentRear * 2 + A[i])%10;
            if(currentRear == 0|| currentRear==5)
                res.add(true);
            else
                res.add(false);
        }
        return res;
    }
}
